src/Features/Core/Portable/CodeFixes/Configuration/ConfigureSeverityLevelCodeFixProvider.ConfigurationUpdater.cs

```diff
@@ -0,0 +1,357 @@
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.CodeAnalysis.Options;
+using System;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CodeFixes.Configuration
+{
+    internal sealed partial class ConfigureSeverityLevelCodeFixProvider : ISuppressionOrConfigurationFixProvider
+    {
+        private sealed partial class ConfigurationUpdater
+        {
+            private const string DiagnosticOptionPrefix = "dotnet_diagnostic.";
+            private const string DiagnosticOptionSuffix = ".severity";
+
+            // Regular expression for .editorconfig header.
+            // For example: "[*.cs]    # Optional comment"
+            private static readonly Regex s_headerPattern = new Regex(@"\[\*([^#\[]*)\]([ ]*[;#].*)?");
+
+            // Regular expression for .editorconfig code style option entry.
+            // For example: "dotnet_style_object_initializer = true:suggestion   # Optional comment"
+            private static readonly Regex s_optionBasedEntryPattern = new Regex(@"([\w ]+)=([\w ]+):[ ]*([\w]+)([ ]*[;#].*)?");
+
+            // Regular expression for .editorconfig diagnosticID severity configuration entry.
+            // For example: "dotnet_diagnostic.CA2000.severity = suggestion   # Optional comment"
+            private static readonly Regex s_severityBasedEntryPattern = new Regex(@"([\w \.]+)=[ ]*([\w]+)([ ]*[;#].*)?");
+
+            private readonly string _severity;
+            private readonly Diagnostic _diagnostic;
+            private readonly Project _project;
+            private readonly CancellationToken _cancellationToken;
+            private readonly string _language;
+
+            private ConfigurationUpdater(
+                string severity,
+                Diagnostic diagnostic,
+                Project project,
+                CancellationToken cancellationToken)
+            {
+                _severity = severity;
+                _diagnostic = diagnostic;
+                _project = project;
+                _cancellationToken = cancellationToken;
+                _language = project.Language;
+            }
+
+            /// <summary>
+            /// Updates or adds an .editorconfig <see cref="AnalyzerConfigDocument"/> to the given <paramref name="project"/>
+            /// so that the severity of the given <paramref name="diagnostic"/> is configured to be the given
+            /// <paramref name="severity"/>.
+            /// </summary>
+            public static async Task<Solution> ConfigureEditorConfig(
+                string severity,
+                Diagnostic diagnostic,
+                Project project,
+                CancellationToken cancellationToken)
+            {
+                var updater = new ConfigurationUpdater(severity, diagnostic, project, cancellationToken);
+                return await updater.Configure().ConfigureAwait(false);
```

Comment:
async/await can all be removed from this static helper method.

Reply:
Done. A code fix to detect and auto-fix this would be nice :-)

Reply:
i agree.  though there's a slight and subtle change in semantics that can occasionally be impactful.  It's not an issue here... but it's something we might get dinged on if it impacts someone negatively. :)